server/server.go

```diff
@@ -519,12 +520,56 @@ func (server HTTPServer) singleRuleEndpoint(writer http.ResponseWriter, request
 		return
 	}
 
+	if rule.Internal == true {
+		ok := server.checkInternalRulePermissions(writer, request)
+		if ok != true {
+			// handled in function
+			return
+		}
+	}
+
 	err = responses.SendOK(writer, responses.BuildOkResponseWithData("report", rule))
 	if err != nil {
 		log.Error().Err(err).Msg(responseDataError)
 	}
 }
 
+// checkInternalRulePermissions checks if organizations for internal rules are enabled
+// if so, retrieves the org_id from request/token and returns whether that ID is on the list
+// of allowed organizations to access internal rules
+func (server HTTPServer) checkInternalRulePermissions(writer http.ResponseWriter, request *http.Request) bool {
+	if !server.Config.EnableInternalRulesOrganizations || !server.Config.Auth {
+		return true
```

Comment:
Does it mean that if internal rules are disabled, user would still get them or I'm missing something?

Reply:
Ah, got it, it's about restricting internal rules to some organizations from the list, not the rules at all :)

Reply:
yeah, but in reality, the filter will be disabled for most time AFAIK